code/datums/supplypacks.dm

```diff
@@ -1453,6 +1455,22 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
 	containername = "supermatter shard crate"
 	group = "Engineering"
 	access = list(access_engine_equip)
+	var/static/list/shard_counts_by_user = list()
+
+
+/datum/supply_packs/supermatter_shard/OnConfirmed(var/mob/user)
+	shard_counts_by_user[user.ckey]++
+	var/i = shard_counts_by_user[user.ckey]
+	var/span = ""
+	switch (i)
+		if (1)
+			span = "notice"
+		if (2 to 5)
+			span = "warning"
+		else
+			span = "danger"
+	message_admins("<span class='[span]'>[key_name(user)] has ordered a supermatter shard supplypack, this is his #[i] order. @[formatJumpTo(user)]</span>")
+	log_admin("[key_name(user)] has ordered a supermatter shard supplypack, this is his #[i] order. @([user.x], [user.y], [user.z])")
```

Comment:
Doesn't `message_admins` also write to logs?

Reply:
It does, but I see this duo everywhere in the codebase
If I have to defend it in a lazy way, the `log_admin` has less `<span>` bloat and that makes it easier to review in a ban request